theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -460,30 +466,55 @@ def _optimize_impl(
     def compute_delta(self, **kwargs) -> torch.Tensor:
         pass
 
+    # Given descent directions and step sizes, updates the optimization
+    # variables.
+    # Batch indices indicated by `converged_indices` mask are ignored
+    # unless `force_update = True`.
+    # Returns the total error tensor after the update
+    def step(
+        self,
+        delta: torch.Tensor,
+        previous_err: torch.Tensor,
+        converged_indices: torch.Tensor,
+        force_update: bool,
+        **kwargs,
+    ) -> torch.Tensor:
+        # makes sure tmp cotainers are up to date with current variables
+        self._update_tmp_optim_vars()
+        # stores the result of the retract step in `self._tmp_optim_vars`
+        self.objective.retract_vars_sequence(
```

Comment:
Slightly confused, would this update the optim vars of the objective with the delta retract on the tmp optim vars? Or does this happen below when calling objective.update.

Reply:
Okay. Maybe a brief comment block at the top might be helpful to write more optimizers in the future: describing a typical flow of what functions get called, in what order, and what they do.

Reply:
Good idea, I'll add it.

Reply:
Added.